notebook/notebookapp.py

```diff
@@ -238,6 +238,20 @@ def init_handlers(self, settings):
         handlers.extend(load_handlers('services.nbconvert.handlers'))
         handlers.extend(load_handlers('services.kernelspecs.handlers'))
         handlers.extend(load_handlers('services.security.handlers'))
+        
+        # BEGIN HARDCODED WIDGETS HACK
+        try:
+            import ipywidgets
+            handlers.append(
+                (r"/nbextensions/widgets/(.*)", FileFindHandler, {
+                    'path': ipywidgets.find_static_assets(),
+                    'no_cache_paths': ['/'], # don't cache anything in nbextensions
+                }),
+            )
+        except:
+            ipython_app.log.warn('ipywidgets package not installed.  Widgets are unavailable.')
```

Comment:
I think this can be `app_log`, and you don't need to add `ipython_app`as an arg to this method.